Source/WebKit/UIProcess/ProcessAssertion.cpp

```diff
@@ -52,6 +53,32 @@ ASCIILiteral processAssertionTypeDescription(ProcessAssertionType type)
     return "unknown"_s;
 }
 
+Ref<ProcessAssertion> ProcessAssertion::create(ProcessID processID, const String& reason, ProcessAssertionType type, Mode mode, const String& environmentIdentifier, CompletionHandler<void()>&& acquisisionHandler)
+{
+    auto assertion = adoptRef(*new ProcessAssertion(processID, reason, type, environmentIdentifier));
+    if (mode == Mode::Async)
+        assertion->acquireAsync(WTFMove(acquisisionHandler));
+    else {
+        assertion->acquireSync();
+        if (acquisisionHandler)
+            acquisisionHandler();
+    }
+    return assertion;
+}
+
+Ref<ProcessAssertion> ProcessAssertion::create(AuxiliaryProcessProxy& process, const String& reason, ProcessAssertionType type, Mode mode, CompletionHandler<void()>&& acquisisionHandler)
+{
+    auto assertion = adoptRef(*new ProcessAssertion(process, reason, type));
+    if (mode == Mode::Async)
```

Comment:
All the code from here on is identical to the `ProcessID` version of the create method. I think you should move this common initialization code into a helper (maybe `createCommon` or something) that would share that logic.

Reply:
Good catches! I have resolved these issues in the latest patch.

Thanks for reviewing!